### ISSUE TITLE
Publish Mapbox Android 2.0.0

### ISSUE THREAD

**Issue by bleege** [mapbox/mapbox-gl-native#2515]
Release Mapbox Android 2.0.0 to Maven Central.  This release will contain:

* [`android-v0.1.0`](https://github.com/mapbox/mapbox-gl-native/tree/android-v0.1.0) tag
* Cherry Pick [`Set Android To 2.0.0`](https://github.com/mapbox/mapbox-gl-native/issues/2486)

/cc @mapbox/mobile @twbell @tobrun @katydecorah @tatsvc @jfirebaugh

**Comment by bleege** [comment]
I just created [`release-android-v2.0.0`](https://github.com/mapbox/mapbox-gl-native/tree/release-android-v2.0.0) branch based off of the [`android-v0.1.0`](https://github.com/mapbox/mapbox-gl-native/tree/android-v0.1.0) tag.

Next up is to start Cherry Picking and then doing a 2.0.0-SNAPSHOT test to make sure everything is :ok_hand: .

**Comment by incanus** [comment]
Confirming: cherry-picking rename/restructure-related commits from `master`, right? That was `android-v0.1.0 == android-v2.0.0` source code.

**Comment by bleege** [comment]
Confirmed:

1. Cherry picking from `master` to `release-android-v2.0.0`.  
1. `release-android-v2.0.0` is `android-v0.1.0` source code.

**Comment by bleege** [comment]
Just finished cherry picking (and painfully merging) the 4 commits from #2486 that do the giant rename process (see below).  Will now attempt to build and then :eyes: things.

* c6d9fda61f4a75d4febca79ce155b446d84cb369
* 212c7ed8da899034561b94bbf0c962644d2885ff
* bf440de7f86481d816bcbd399afc7abee909a9ea
* eb59bccfc1bf95626625fab4d6faf07a1e763a8b

**Comment by bleege** [comment]
Ugh... the build failed in JNI-land.  Pondering re-implementing changes in IDE (as original rename process did in #2486) as it'll likely be more accurate than Cherry Picking commits from `master`'s distant future.

![screen shot 2015-10-02 at 3 45 00 pm](https://cloud.githubusercontent.com/assets/989928/10257382/9a9b48a6-691c-11e5-8709-e424a8dd6b2c.png)

**Comment by bleege** [comment]
Tried cherry picking a 2nd time and got the same `renderSync()` error as above.  Going to try recreating the 5 commits from #2486 locally from the last `release-android-v2.0.0` commit.

 
![screen shot 2015-10-02 at 6 07 01 pm](https://cloud.githubusercontent.com/assets/989928/10259707/f4cc9686-6930-11e5-9d0f-96731dba9c91.png)
_Failed Attempt 2_

**Comment by bleege** [comment]
Just did a successful `make android` from the last of the `release-android-v2.0.0` branch.  Worked as expected which is good.  Time to reinvent the :ferris_wheel: .

**Comment by bleege** [comment]
Just built 2.0.0-SNAPSHOT 3 based off [`release-android-v2.0.0-redo-working`](https://github.com/mapbox/mapbox-gl-native/tree/release-android-v2.0.0-redo-working).  Will now test internally.

https://oss.sonatype.org/content/repositories/snapshots/com/mapbox/mapboxsdk/mapbox-android-sdk/2.0.0-SNAPSHOT/

**Comment by bleege** [comment]
Internal testing revealed 2 instances of the same error.  This error turned out to be related to an internal app specific code and NOT the SDK.  Based on this, `2.0.0` seems just as stable as the original `0.1.0` release.  Will proceed to release `2.0.0` to Maven Central.

**Comment by bleege** [comment]
I just finished publishing Mapbox Android 2.0.0 to Maven Central.  It's the exact same binaries as the SNAPSHOT 3 from Friday night that is based off of [`release-android-v2.0.0-redo-working`](https://github.com/mapbox/mapbox-gl-native/tree/release-android-v2.0.0-redo-working).

The next steps are to tag this release and then merge [`release-android-v2.0.0-redo-working`](https://github.com/mapbox/mapbox-gl-native/tree/release-android-v2.0.0-redo-working) back into `master`.

Once that's completed we can officially deprecate [Mapbox Android Raster](https://github.com/mapbox/mapbox-android-sdk/).

**Comment by bleege** [comment]
Release has been tagged [`android-v2.0.0`](https://github.com/mapbox/mapbox-gl-native/commits/android-v2.0.0).

Now we merge.

**Comment by bleege** [comment]
I consulted with @incanus @jfirebaugh on the merge issue in #2525 and together we decided not to merge [`release-android-v2.0.0-redo-working`](https://github.com/mapbox/mapbox-gl-native/tree/release-android-v2.0.0-redo-working) branch back into `master` as it is redundant.  For the sake of any possible `2.0.x` releases we will use the [`android-v2.0.0`](https://github.com/mapbox/mapbox-gl-native/commits/android-v2.0.0) tag to create a development branch and then tag it when released.  `2.1.x` releases will be from `master` as it is today.

I also have confirmed that Mapbox Android 2.0.0 is now on Maven Central and accessible to developers.  Taken with the results of the tagging / branching strategy this means that this ticket can now be closed.
